tests/test_slurm_settings.py

```diff
@@ -245,6 +247,40 @@ def test_format_env_vars():
     assert all("SSKEYIN" not in x for x in formatted)
 
 
+def test_catch_existing_env_var(caplog):
+    rs = SrunSettings(
+        "python",
+        env_vars={
+            "SMARTSIM_TEST_VAR": "B",
+        },
+    )
+    os.environ["SMARTSIM_TEST_VAR"] = "A"
+    os.environ["SMARTSIM_TEST_CSVAR"] = "A,B"
+    caplog.clear()
+    rs.format_env_vars()
+
+    msg = f"Variable SMARTSIM_TEST_VAR is set to A in current environment. "
+    msg += f"If the job is running in an interactive allocation, the value B will not be set. "
+    msg += "Please consider removing the variable from the environment and re-running the experiment."
+
+    for record in caplog.records:
+        assert record.levelname == "WARNING"
+        assert record.message == msg
+
+    caplog.clear()
+
+    env_vars = {"SMARTSIM_TEST_VAR": "B", "SMARTSIM_TEST_CSVAR": "C,D"}
+    settings = SrunSettings("python", env_vars=env_vars)
+    settings.format_comma_sep_env_vars()
+
+    for record in caplog.records:
+        assert record.levelname == "WARNING"
+        assert record.message == msg
+
+    os.environ.pop("SMARTSIM_TEST_VAR", None)
+    os.environ.pop("SMARTSIM_TEST_CSVAR", None)
```

Comment:
Incredibly minor refactor request, don't feel the need to make this change if you don't want:

If you'd like, `monkeypatch.setenv` will automatically handle the resetting of the envvars back to their original values a the end of a test

(Shouldn't matter unless someone sources pytest AND has `SMARTSIM_TEST_VAR` or `SMARTSIM_TEST_CSVAR` set in their env, so the chances of this mattering are minuscule)

Reply:
Thanks for introducing me to the joys of `monkeypatch`.